openmdao/core/indepvarcomp.py

```diff
@@ -158,11 +156,12 @@ def add_output(self, name, val=1.0, shape=None, units=None, res_units=None, desc
         if res_ref is None:
             res_ref = ref
 
-        if tags is not None and not isinstance(tags, (str, list)):
-            raise TypeError('The tags argument should be a str or list')
-
-        tags = convert_user_defined_tags_to_set(tags)
-        tags.add('indep_var')
+        if tags is None:
+            tags = {'indep_var'}
+        else:
+            if not isinstance(tags, (str, set, list)):
```

Comment:
Do we need this check since make_set will raise its own exception if tags is not of the proper type?

Reply:
removed and added arg to `make-set` to enable user-friendlier error message